packages/ui/src/hooks/useAccounts.ts

```diff
@@ -9,25 +9,20 @@ interface UseAccounts {
 
 export function useAccounts(): UseAccounts {
   const keyring = useKeyring()
-  const [state, setState] = useState<UseAccounts>({ allAccounts: [], hasAccounts: false })
+  const accounts = useObservable(keyring.accounts.subject.asObservable(), [keyring])
 
-  useEffect(() => {
-    const subscription = keyring.accounts.subject.subscribe((accounts): void => {
-      const allAccounts = accounts
-        ? Object.values(accounts).map((account) => ({
-            address: account.json.address,
-            name: account.json.meta.name,
-          }))
-        : []
-      const hasAccounts = allAccounts.length !== 0
+  const allAccounts: Account[] = []
 
-      setState({ allAccounts, hasAccounts })
-    })
+  if (accounts) {
+    allAccounts.push(
+      ...Object.values(accounts).map((account) => ({
+        address: account.json.address,
+        name: account.json.meta.name,
+      }))
+    )
+  }
 
-    return () => {
-      subscription.unsubscribe()
-    }
-  }, [keyring])
+  const hasAccounts = allAccounts.length !== 0
 
-  return state
+  return { allAccounts, hasAccounts }
```

Comment:
I'm a bit concerned that we return `allAccounts`, which a) is not a state and b) is appended with keyring accounts. But apparently the code in this hook is invoked only once, so it works fine...

My expectation would be for `accounts` to be updated whenever something is changed in the extension (eg. an account added or deleted), but looks like it does not work like this.

Reply:
I'm planning to refactor this together with #64 :) I've already looked at it and it looks like we could use that cheaply.  
I've added this short PR as I saw that, at least on my machine, accounts were not always added from the extension.